working-with-openai/open-ai-api-in-projects.md

```diff
@@ -0,0 +1,381 @@
+# Using the OpenAI API
+
+## Goals
+
+Now that we know a bit about the OpenAI endpoints, lets learn how to use them within our programs. To do so, we will work with a partially finished API that creates video game NPCs (Non Playable Characters). Within that program, we will install the OpenAI library and then leverage one of OpenAI's endpoints to generate dialogue for characters we create.
+
+Our goals for this lesson are to:
+- Integrate the OpenAI library into a python project.
+- Use the OpenAI chat completion endpoint to generate content for our project. 
+
+## Set Up the NPC Generator
+
+This tutorial will walk you through how to integrate the OpenAI API into your projects with the companion repo [NPC Generator](https://github.com/AdaGold/NPC-Generator). We'll start with our usual setup.
+
+1. Fork the repo to your github and then clone to your machine.
+2. Create and run a virtual environment for the project.
+3. Use pip to install the requirements.txt.
+
+## Explore the NPC Generator
+
+Take a second to explore the NPC Generator in its current state. For this project, we will be finishing an API that creates NPCs (Non Playable Characters) for a Witcher style game. We currently have two models, "Character" and "Greeting". "Character" holds some basic information about a character and has a one to many relationship with "Greeting", which represents a stock phrase that particular character may say when interacting with the player. Each NPC can have multiple greetings but each greeting can only be connected to one character. We will use the OpenAI API to generate those greetings which we will then store in our database.
```

Comment:
To make the intro as accessible as possible, I want to generalize the idea of what we're building since many folks may not know The Witcher. 

```suggestion
## Explore the NPC Generator repo

Take some time to explore the NPC Generator repo in its current state. For this project, we will be finishing an API for a role-playing style game. Our API creates NPCs (Non Playable Characters) that your player's character will interact with. 

We currently have two models, "Character" and "Greeting". "Character" holds some basic information about a character and has a one to many relationship with "Greeting", which represents a stock phrase that particular character may say when interacting with the player. Each NPC can have multiple greetings but each greeting can only be connected to one character. We will use the OpenAI API to generate those greetings which we will then store in our database.
```